src/unix/sentry_unix_path.c

```diff
@@ -20,6 +25,50 @@ struct sentry_pathiter_s {
     DIR *dir_handle;
 };
 
+sentry_path_t *
+sentry__path_current_exe(void)
+{
+#ifdef SENTRY_PLATFORM_DARWIN
+    // inspired by:
+    // https://github.com/rust-lang/rust/blob/0176a9eef845e7421b7e2f7ef015333a41a7c027/src/libstd/sys/unix/os.rs#L339-L358
+    uint32_t buf_size = 0;
+    _NSGetExecutablePath(NULL, &buf_size);
+    char *buf = sentry_malloc(buf_size);
+    if (!buf) {
+        return NULL;
+    }
+    int err = _NSGetExecutablePath(buf, &buf_size);
+    if (err) {
+        sentry_free(buf);
+        return NULL;
+    }
+    return sentry__path_from_str_owned(buf);
+#elif defined(SENTRY_PLATFORM_LINUX)
+    // inspired by:
+    // https://github.com/rust-lang/rust/blob/0176a9eef845e7421b7e2f7ef015333a41a7c027/src/libstd/sys/unix/os.rs#L328-L337
+    char buf[4096];
+    readlink("/proc/self/exe", buf, 4096);
+    return sentry__path_from_str(buf);
+#endif
+    return NULL;
+}
+
+sentry_path_t *
+sentry__path_dir(sentry_path_t *path)
```

Comment:
Since this internally modifies the path, shouldn't you pass a `const sentry_path_t *path` and clone it internally?